prisma-fmt/src/hover.rs

```diff
@@ -0,0 +1,129 @@
+use log::info;
+use lsp_types::{Hover, HoverContents, HoverParams, MarkupContent, MarkupKind};
+use psl::{
+    error_tolerant_parse_configuration,
+    parser_database::ParserDatabase,
+    schema_ast::ast::{self, FieldPosition},
+    Diagnostics, SourceFile,
+};
+
+use crate::{offsets::position_to_offset, LSPContext};
+
+pub(super) type HoverContext<'a> = LSPContext<'a, HoverParams>;
+
+impl<'a> HoverContext<'a> {
+    pub(super) fn position(&self) -> Option<usize> {
+        let pos = self.params.text_document_position_params.position;
+        let initiating_doc = self.initiating_file_source();
+
+        position_to_offset(&pos, initiating_doc)
+    }
+}
+
+pub fn run(schema_files: Vec<(String, SourceFile)>, params: HoverParams) -> Hover {
+    let (_, config, _) = error_tolerant_parse_configuration(&schema_files);
+
+    let db = {
+        let mut diag = Diagnostics::new();
+        ParserDatabase::new(&schema_files, &mut diag)
+    };
+
+    let Some(initiating_file_id) = db.file_id(params.text_document_position_params.text_document.uri.as_str()) else {
+        info!("Initiating file name is not found in the schema");
+        panic!("Initiating file name is not found in the schema");
+    };
+
+    let ctx = HoverContext {
+        db: &db,
+        config: &config,
+        initiating_file_id,
+        params: &params,
+    };
+
+    hover(ctx)
+}
+
+fn hover(ctx: HoverContext<'_>) -> Hover {
```

Comment:
`hover` should return an `Option<Hover>` instead of instantiating empty `Hover`'s